src/parser/druid/restoration/modules/features/PrematureRejuvenations.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS/index';
+import Analyzer from 'parser/core/Analyzer';
+import HealingDone from 'parser/shared/modules/HealingDone';
+import StatisticBox from 'interface/others/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import Combatants from 'parser/shared/modules/Combatants';
+
+const debug = false;
+
+const REJUV_DURATION = 15000;
+const MS_BUFFER = 200;
+const PENDEMIC_THRESHOLD = 0.7;
+const FLOURISH_EXTENSION = 8000;
+
+/*
+ * This module tracks early refreshments of rejuvenation.
+ * TODO: Extend/refactor this module to include other HoTs/Spells as well such as lifebloom/efflorescence
+ * TODO: Add this module to checklist
+ */
+class PrematureRejuvenations extends Analyzer {
+  static dependencies = {
+    healingDone: HealingDone,
+    combatants: Combatants,
+  };
+
+  totalRejuvsCasts = 0;
+  rejuvenations = [];
+  earlyRefreshments = 0;
+  timeLost = 0;
+  pandemicExtensions = 0;
+  pandemicExtensionsTime = 0;
```

Comment:
What is pandemicExtensionsTime needed for when timeLost exists? Is it really relevant to show how much correct pandemic time you had?

Reply:
I thought it'd be interesting to show/know that information.